lib/ui/popover/popover.js

```diff
@@ -44,12 +44,26 @@
     this.destroy = function() {
       $element.popover('destroy');
     };
+
+    if($scope.showOnLoad) {
+      this.show();
+      if($scope.delay && $scope.delay.hide) {
+        $timeout(this.hide, $scope.delay.hide);
+        return;
+      }
+      // If no delay is found or cannot be parsed, set a default timeout so that the popover doesn't stick around forever
+      $timeout(this.hide, 10000);
```

Comment:
`$timeout(this.hide, 10000, false);` - If set to false skips model dirty checking.  This is typically a performance boost if your function `this.hide` doesn't depend on data that may have changed.  Let me know your thoughts